invenio_jobs/jobs.py

```diff
@@ -102,5 +103,35 @@ def _build_task_arguments(cls, job_obj, since=None, custom_args=None, **kwargs):
             return custom_args
 
         if since is None and job_obj.last_runs["success"]:
-            since = job_obj.last_runs["success"].started_at
+            """
+            The most common case: `since` has not been manually specified by the user, so we
+            set it to the start time of the last successful job.
+
+            We can add the UTC time zone as we store all dates as UTC in the database.
+            For comparison with other dates in job implementors, it's useful to have TZ info in the timestamp.
+            """
+
+            since = job_obj.last_runs["success"].started_at.replace(tzinfo=timezone.utc)
+        elif since is not None:
+            """
+            The user has manually specified `since` as an ISO-format string. If they gave a timezone, we should ensure
+            we translate the timestamp to UTC. If they have not specified a timezone, we assume they meant UTC.
+            """
+            if isinstance(since, str):
```

Comment:
I think that we are doing this check is the wrong place. We do param validation/conversion/transformation when loading via Marshmallow using schema (example [here](https://github.com/inveniosoftware/invenio-jobs/blob/master/invenio_jobs/services/config.py#L137)).